src/services/CasperServiceByJsonRPC.ts

```diff
@@ -345,6 +345,23 @@ export class CasperServiceByJsonRPC {
       .then(res => BigNumber.from(res.balance_value));
   }
 
+  public async queryBalance(
+    purseIdentifier: string,
+    stateRootHash?: string
+  ): Promise<BigNumber> {
+    return await this.client
+      .request({
+        method: 'query_balance',
+        params: {
+          purse_identifier: {
+            main_purse_under_public_key: purseIdentifier
```

Comment:
This looks incorrect to me, the purse_identifier arg is being coerced here to the `main_purse_under_public_key` field, which feels like an incorrect implementation of the RPC call. 

Maybe this is me not reading JS very well, and I wouldn't mind going over this in further detail @hoffmannjan

Reply:
You're right! I superficially read the [PR description](https://github.com/casper-network/casper-node/pull/2868) and took the PR description as API spec, my fault!.

I added a new enum `PurseIdentifier` which has variants  matching the `casper-client-rs` implementation. This version of SDK is dump when it comes to type checking an RPC method arguments (thats why I added new argument to `queryBalance` - `purseIdentifierType`) - but that's something that I want to address in `v3` (it will be breaking change for the users).